ci/test/03_test_script.sh

```diff
@@ -146,21 +146,74 @@ if [ "$RUN_CHECK_DEPS" = "true" ]; then
   "${BASE_ROOT_DIR}/contrib/devtools/check-deps.sh" .
 fi
 
+function get_interfaces()
+{
+  set -o pipefail
+  ifconfig | awk -F ':| ' '/^[^[:space:]]/ { if (!match($1, /^lo/)) { print $1 } }'
+  set +o pipefail
+}
+
+function tcpdump_file()
+{
+  echo "/tmp/tcpdump_$1_$2"
+}
+
+function traffic_monitor_begin()
+{
+  test_name="$1"
+  for ifname in $(get_interfaces) ; do
+    tcpdump -nU -i "$ifname" -w "$(tcpdump_file "$test_name" "$ifname")" &
+  done
+}
+
+function traffic_monitor_end()
+{
+  test_name="$1"
+
+  for ifname in $(get_interfaces) ; do
+    f=$(tcpdump_file "$test_name" "$ifname")
+    if [ ! -e "$f" ] && [ "$CI_FAIL_IF_NO_TCPDUMP_FILE" != "1" ] ; then
+      # In some CI environments this script is not running as root and so the
+      # tcpdump errors and does not create $f. Skip silently those, but we
+      # need at least one where tcpdump can run and this is the ASAN one. So
+      # treat the absence of $f as an error only on the ASAN task.
+      continue
+    fi
```

Comment:
The table in the OP describes each task and its status wrt to this. There are 18 tasks. It is supposed to work on the top 5+4=9 tasks. The other 9 tasks are not expected to run this - e.g. Windows or tidy that does not run any tests or macOS that runs without docker and has no enough permissions.

The intention here is to have at least one task that runs this. But it will not hurt to enforce it on more. Do you want to add `CI_FAIL_IF_NO_TCPDUMP_FILE=1` to more tasks? Or reverse the logic to `CI_OK_IF_NO_TCPDUMP_FILE=1` in the tasks that are expected to not run this (maybe it is the same if it is 9 vs 9 tasks)?